### PR TITLE
find elixir modules that require adding an alias

### PR DESCRIPTION
**Pull request by ajayvigneshk**
Basic code to find modules that require adding an alias. Sort of related to [elixir-ls issue](https://github.com/elixir-lsp/elixir-ls/issues/161)
~Marking it draft due to some known test failures that I would like to get some suggestions on how to handle them.~

### CODE REVIEW

**Comment by ajayvigneshk** [comment]
Two of the tests fail as the newly added code suggests `Kernel.SpecialForms` as a new alias. I'm thinking this can be ignored from suggestions to fix.
One other test failure can be fixed by ignoring the suggestions from the new code(I've used this approach in some tests already, not sure if it's okay though)

File: lib/elixir_sense/core/metadata.ex [hunk 1]
```diff
@@ -58,6 +60,31 @@ defmodule ElixirSense.Core.Metadata do
...
+      Map.has_key?(metadata.moduledoc_positions, module) ->
+        Map.get(metadata.moduledoc_positions, module)
+
+      true ->
+        mod_info = Map.get(metadata.mods_funs_to_positions, {env.module, nil, nil})
+
+        case mod_info do
+          %State.ModFunInfo{positions: [{line, column}]} ->
+            # Hacky :shrug
+            {line + 1, column - 10 + 2}
```
Comment (antedeguemon) [inline lib/elixir_sense/core/metadata.ex:80]: What do these numbers represent?

I think you could extract them to variables, just to get the [magic-number](https://wiki.c2.com/?MagicNumber) look out of it:

```suggestion
            line_offset = 1
            column_offset = -8
            
            {line + line_offset, column + column_offset}
```

File: lib/elixir_sense/providers/suggestion/complete.ex [hunk 2]
```diff
@@ -88,19 +88,19 @@ defmodule ElixirSense.Providers.Suggestion.Complete do
               behaviours: []
   end
 
-  @spec complete(String.t(), Env.t()) ::
+  @spec complete(String.t(), Env.t(), list(keyword())) ::
```
Comment (lukaszsamson) [inline lib/elixir_sense/providers/suggestion/complete.ex:91]:
```suggestion
  @spec complete(String.t(), Env.t(), keyword()) ::
```

File: lib/elixir_sense/providers/suggestion.ex [hunk 3]
```diff
@@ -99,12 +99,14 @@ defmodule ElixirSense.Providers.Suggestion do
     bitstring_options: &Reducers.Bitstring.add_bitstring_options/5
   ]
 
+  @add_opts_for [:populate_common]
+
   @doc """
   Finds all suggestions for a hint based on context information.
   """
   @spec find(String.t(), State.Env.t(), Metadata.t(), cursor_context, ModuleStore.t()) ::
```
Comment (lukaszsamson) [inline lib/elixir_sense/providers/suggestion.ex:107]:
```suggestion
  @spec find(String.t(), State.Env.t(), Metadata.t(), cursor_context, ModuleStore.t(), keyword()) ::
```

File: lib/elixir_sense.ex [hunk 4]
```diff
@@ -203,7 +203,7 @@ defmodule ElixirSense do
         spec: "@spec insert_at(list, integer, any) :: list", summary: "Returns a list with `value` inserted at the specified `index`."}]
   """
   @spec suggestions(String.t(), pos_integer, pos_integer) :: [Suggestion.suggestion()]
```
Comment (lukaszsamson) [inline lib/elixir_sense.ex:205]:
```suggestion
  @spec suggestions(String.t(), pos_integer, pos_integer, keyword()) :: [Suggestion.suggestion()]
```

File: lib/elixir_sense/core/metadata.ex [hunk 5]
```diff
@@ -28,7 +28,9 @@ defmodule ElixirSense.Core.Metadata do
             types: %{},
             specs: %{},
             structs: %{},
-            error: nil
+            error: nil,
+            first_alias_positions: nil,
```
Comment (lukaszsamson) [inline lib/elixir_sense/core/metadata.ex:32]: Please add those fields to typespec

**Comment by ajayvigneshk** [comment]
@antedeguemon @lukaszsamson I've applied the suggestions and removed the usage of `then` due to failures for v 1.11.

**Comment by ajayvigneshk** [comment]
Made a commit to fix credo errors

**Pull request merged**